File: 백준/Silver/1874. 스택 수열/스택 수열.py
from sys import *
from collections import * 
from heapq import * 
from functools import *
from math import *


n = int(stdin.readline()[:-1])

NO = "NO"

flag = False
curr = 1 

stack = []
ans = []

for _ in range(n):
    m = int(stdin.readline()[:-1])
    
    while curr <= m:
        stack.append(curr)
        ans.append("+")
        curr += 1
        
    if stack[-1] == m:
        stack.pop()
        ans.append("-")
    
    else:
        flag = True
        break

    

if flag:
    print(NO)

else :
    for s in ans:
        print(s)
        

File: 백준/Silver/1874. 스택 수열/스택 수열.swift
let n = Int(readLine()!)!

var stk : [Int] = []

var flag: Bool = true
// 1 ~ n 방향으로

var curr = 1

var ans: [String] = []

for _ in 0..<n {
    let target = Int(readLine()!)! // 4
    
    while curr <= target {
        ans.append("+")
        stk.append(curr)
        curr += 1
    }
    // stk: 1 ~ target
     
    // 1,2,3
    if !stk.isEmpty && stk.last! == target {
        let _ = stk.popLast()
        ans.append("-")
    } else {
        flag = false
        break
    }
    
    // 1,2,3
    
    
}

if flag == false {
    print("NO")
} else {
    print(ans.joined(separator: "\n")) // ans = ["+","-"]
}
